proc/fcidump.py

```diff
@@ -0,0 +1,460 @@
+    be converted to the ordering used in psi4
+    """
+    intdump = {}
+    with open(fname, 'r') as handle:
+        assert '&FCI' == handle.readline().strip()
+
+        skiplines = 1
+        read = True
+        while True:
+            skiplines += 1
+            line = handle.readline()
+            if 'END' in line:
+                break
+
+            key, value = line.split('=')
+            value = value.strip().rstrip(',')
+            if key == 'UHF':
+                value = 'TRUE' in value
+            elif key == 'ORBSYM':
+                value = [int(x) for x in value.split(',')]                
+            elif key == 'PNTGRP':
+                pass
+            else:
+                value = int(value.replace(',', ''))
+
+            intdump[key.lower()] = value
+
+    if convert_to_psi4 and ('pntgrp' in intdump) and ('orbsym' in intdump):
+        irrep_map_inverse = _irrep_map_inverse(intdump['pntgrp'])
+        psi4_irrep_map = map(lambda x: irrep_map_inverse[x], intdump['orbsym'])
+        intdump['orbsym'] = list(psi4_irrep_map)
+
+    # Read the data and index, skip header
+    raw_ints = np.genfromtxt(fname, skip_header=skiplines)
+
+    # Read last line, i.e. Enuc + Efzc
+    intdump['enuc'] = raw_ints[-1, 0]
+
+    # Read in integrals and indices
+    ints = raw_ints[:-1, 0]
+
+    # Get dimensions and indices
+    nbf = intdump['norb']
+    idxs = raw_ints[:, 1:].astype(np.int) - 1
+
+    # Slices
+    sl = slice(ints.shape[0] - nbf, ints.shape[0])
+
+    # Count how many 1-index intdump we have
+    one_index = np.all(idxs[sl, 1:] == -1, axis=1).sum()
+
+    # Extract orbital energies if present
+    if one_index > 0:
+        epsilon = np.zeros(nbf)
+        epsilon[idxs[sl, 0]] = ints[sl]
+        intdump['epsilon'] = epsilon
+
+    # Count how many 2-index intdump we have
+    sl = slice(ints.shape[0] - one_index - nbf * nbf, sl.stop - one_index)
+    two_index = np.all(idxs[sl, 2:] == -1, axis=1).sum()
+    sl = slice(ints.shape[0] - two_index - one_index, ints.shape[0] - one_index)
+
+    # Extract Hcore
+    Hcore = np.zeros((nbf, nbf))
+    Hcore[(idxs[sl, 0], idxs[sl, 1])] = ints[sl]
+    Hcore[(idxs[sl, 1], idxs[sl, 0])] = ints[sl]
+    intdump['hcore'] = Hcore
+
+    # Extract ERIs
+    sl = slice(0, sl.start)
+    eri = np.zeros((nbf, nbf, nbf, nbf))
+    eri[(idxs[sl, 0], idxs[sl, 1], idxs[sl, 2], idxs[sl, 3])] = ints[sl]
+    eri[(idxs[sl, 0], idxs[sl, 1], idxs[sl, 3], idxs[sl, 2])] = ints[sl]
+    eri[(idxs[sl, 1], idxs[sl, 0], idxs[sl, 2], idxs[sl, 3])] = ints[sl]
+    eri[(idxs[sl, 1], idxs[sl, 0], idxs[sl, 3], idxs[sl, 2])] = ints[sl]
+    eri[(idxs[sl, 2], idxs[sl, 3], idxs[sl, 0], idxs[sl, 1])] = ints[sl]
+    eri[(idxs[sl, 3], idxs[sl, 2], idxs[sl, 0], idxs[sl, 1])] = ints[sl]
+    eri[(idxs[sl, 2], idxs[sl, 3], idxs[sl, 1], idxs[sl, 0])] = ints[sl]
+    eri[(idxs[sl, 3], idxs[sl, 2], idxs[sl, 1], idxs[sl, 0])] = ints[sl]
+    intdump['eri'] = eri
```

Comment:
I was wondering if something like DASK should be used here since the ERI can be very large.

Reply:
This is just code from psi4. I don't think that FCIDUMP is the format one would use for very large orbital spaces (probably good for < 100 MOs)

Reply:
Thanks for the information. I did not know that.